pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -153,10 +153,9 @@ def on_exception(self, exception):
 
         This will cause the stream to exit loudly.
         """
-        # If this is DEADLINE_EXCEEDED, then we want to retry.
-        # That entails just returning None.
-        deadline_exceeded = grpc.StatusCode.DEADLINE_EXCEEDED
-        if getattr(exception, 'code', lambda: None)() == deadline_exceeded:
+        # If this is in the list of idempotent exceptions, then we want to
+        # retry. That entails just returning None.
+        if isinstance(exception, self._IDEMPOTENT_RETRIES):
```

Comment:
I hate to be *that girl*, but should you be accessing a private member of the superclass? (no need to do anything here if you think it's fine because it probably is fine)

Reply:
I think it's fine. Internal to the "implementation", not internal to the particular implementation (i.e. class)